lib/src/git_backend.rs

```diff
@@ -1815,6 +1876,55 @@ mod tests {
         assert_eq!(std::str::from_utf8(&sig.data).unwrap(), commit_str);
     }
 
+    #[test]
+    fn round_trip_change_id_via_git_header() {
+        let settings = user_settings_with_change_id();
+        let temp_dir = new_temp_dir();
+        let store_path = temp_dir.path();
+
+        let temp_dir2 = new_temp_dir();
+        let empty_store = temp_dir2.path();
+
+        let git_repo_path = temp_dir.path().join("git");
+        let git_repo = git_init(git_repo_path);
+
+        let signature = gix::actor::Signature {
+            name: GIT_USER.into(),
+            email: GIT_EMAIL.into(),
+            time: gix::date::Time::now_utc(),
+        };
+        let empty_tree_id =
+            gix::ObjectId::from_hex(b"4b825dc642cb6eb9a060e54bf8d69288fbee4904").unwrap();
+
+        let original_change_id = ChangeId::from_hex("c64ee0b6e16777fe53991f9281a6cd25");
+        let commit = gix::objs::Commit {
+            tree: empty_tree_id,
+            parents: smallvec::SmallVec::new(),
+            author: signature.clone(),
+            committer: signature.clone(),
+            encoding: None,
+            message: "git commit message".into(),
+            extra_headers: Vec::from([(
+                BString::new(CHANGE_ID_COMMIT_HEADER.to_vec()),
+                BString::new(original_change_id.reverse_hex().into()),
+            )]),
+        };
+
+        let git_commit_id = git_repo.write_object(&commit).unwrap();
```

Comment:
nit: write this using the `GitBackend` instead of the Gitoxide repo object so we test the write path